components/calendar/src/provider.rs

```diff
@@ -27,6 +27,7 @@ pub struct EraStartDate {
     pub day: u8,
 }
 
+//#[allow(clippy::panic)] // TODO(#NNN) Clippy exceptions need docs or fixing.
 #[icu_provider::data_struct(JapaneseErasV1Marker = "calendar/japanese@1")]
```

Comment:
My guess is that there is an `unwrap` in the proc_macro `icu_provider::data_struct`

Reply:
```diff
diff --git a/components/calendar/src/lib.rs b/components/calendar/src/lib.rs
index df3c2aa29..2ee4cbe15 100644
--- a/components/calendar/src/lib.rs
+++ b/components/calendar/src/lib.rs
@@ -6,7 +6,7 @@
     clippy::unwrap_used,
     clippy::expect_used,
     // If enabled it fails on provider.rs line 31
-    //clippy::panic
+    clippy::panic
 )]
 #![cfg_attr(not(any(test, feature = "std")), no_std)]
 
diff --git a/provider/macros/src/lib.rs b/provider/macros/src/lib.rs
index ce71709d1..caca75fb3 100644
--- a/provider/macros/src/lib.rs
+++ b/provider/macros/src/lib.rs
@@ -111,6 +111,7 @@ fn data_struct_impl(attr: AttributeArgs, item: ItemStruct) -> TokenStream2 {
                     }
                 ));
             }
+            #[allow(clippy::panic)]
             _ => {
                 panic!("Invalid attribute to #[data_struct]")
             }
```

I'm still getting the error locally, but the suppressions need to be added inside the macro. Macros expand the code in place, so clippy rule is being applied to the expanded code, but it's being reported at where the proc_macro is defined.